File: src/main/java/io/zeitwert/fm/portfolio/service/api/PortfolioEvaluationService.java
package io.zeitwert.fm.portfolio.service.api;

import io.zeitwert.fm.portfolio.model.ObjPortfolio;
import io.zeitwert.fm.portfolio.service.api.dto.PortfolioEvaluationResult;

public interface PortfolioEvaluationService {

	/**
	 * Get the accumulated cost projection for a given portfolio
	 * 
	 * @param portfolio the portfolio
	 * @return portfolio evaluation
	 */
	PortfolioEvaluationResult getEvaluation(ObjPortfolio portfolio);

}

File: fm-domain/src/main/java/io/zeitwert/fm/portfolio/api/PortfolioEvaluationService.kt
package io.zeitwert.fm.portfolio.api

import io.zeitwert.fm.portfolio.model.ObjPortfolio
import io.zeitwert.fm.portfolio.api.dto.PortfolioEvaluationResult

interface PortfolioEvaluationService {

	/**
	 * Get the accumulated cost projection for a given portfolio
	 *
	 * @param portfolio the portfolio
	 * @return portfolio evaluation
	 */
	fun getEvaluation(portfolio: ObjPortfolio): PortfolioEvaluationResult

}
